src/secure_channel/mod.rs

```diff
@@ -130,27 +130,36 @@ pub struct EstablishedSecureChannel {
     encryption_nonce: SecureChannelNonce,
     #[zeroize(skip)]
     decryption_nonce: SecureChannelNonce,
-    key: Box<[u8; 32]>,
+    /// The key that we will use to encrypt messages
+    encryption_key: Box<[u8; 32]>,
+    /// The key that the other party will use to encrypt messages
+    decryption_key: Box<[u8; 32]>,
     #[zeroize(skip)]
     check_code: CheckCode,
+    /// Whether we initiated the secure channel or not. i.e. are we Device G?
+    initiator: bool,
 }
 
+const CHECKCODE_INFO_STRING: &str = "MATRIX_QR_CODE_LOGIN_CHECKCODE";
+const ENCKEY_S_INFO_STRING: &str = "MATRIX_QR_CODE_LOGIN_ENCKEY_S";
+const ENCKEY_G_INFO_STRING: &str = "MATRIX_QR_CODE_LOGIN_ENCKEY_G";
```

Comment:
I like constants to be either completely local, or if they aren't scoped to a `impl` block or even function, at the top of the file. They are now between a struct and its `impl` block which I find to be the last place I would expect them to be.

Though don't worry, I'll move them where I'll want them.